core/src/main/java/io/questdb/cairo/TxnScoreboardV1.java

```diff
@@ -0,0 +1,257 @@
+    private TableToken tableToken;
+
+    public TxnScoreboardV1(FilesFacade ff, int entryCount, TableToken tableToken) {
+        this.ff = ff;
+        this.pow2EntryCount = Numbers.ceilPow2(entryCount);
+        this.size = TxnScoreboardV1.getScoreboardSize(pow2EntryCount);
+        this.tableToken = tableToken;
+    }
+
+    public static native long getScoreboardSize(int entryCount);
+
+    @Override
+    public boolean acquireTxn(int id, long txn) {
+        assert txn > -1;
+        final long internalTxn = toInternalTxn(txn);
+        final long response = acquireTxn(mem, internalTxn);
+        if (response == 0) {
+            // all good
+            return true;
+        }
+        if (response == -1) {
+            // retry
+            return false;
+        }
+        final long min = fromInternalTxn(-response - 2);
+        throw CairoException.critical(0).put("max txn-inflight limit reached [txn=").put(txn)
+                .put(", min=").put(min)
+                .put(", size=").put(pow2EntryCount)
+                .put(']');
+    }
+
+    @Override
+    public void close() {
+        if (mem != 0) {
+            ff.munmap(mem, size, MemoryTag.MMAP_DEFAULT);
+            mem = 0;
+        }
+
+        if (ff.close(fd)) {
+            LOG.debug().$("closed [fd=").$(fd).I$();
+            fd = -1;
+        }
+    }
+
+    @TestOnly
+    public long getActiveReaderCount(long txn) {
+        return getCount(mem, toInternalTxn(txn));
+    }
+
+    @Override
+    public int getEntryCount() {
+        return pow2EntryCount;
+    }
+
+    @TestOnly
+    public long getMin() {
+        final long min = getMin(mem);
+        // min can be 0 on empty scoreboard, so we simply treat it as txn 0.
+        if (min == 0) {
+            return 0;
+        }
+        return fromInternalTxn(min);
+    }
+
+    @Override
+    public TableToken getTableToken() {
+        return tableToken;
+    }
+
+    @Override
+    public boolean hasEarlierTxnLocks(long maxTxn) {
+        try {
+            if (acquireTxn(0, maxTxn)) {
+                releaseTxn(0, maxTxn);
+            }
+            return getMin() < maxTxn;
+        } catch (CairoException ex) {
+            // Scoreboard can be over allocated, don't stall writing because of that.
+            // Schedule async purge and continue
+            LOG.critical().$("cannot lock last txn in scoreboard, partition purge will be scheduled [table=")
+                    .$(tableToken)
+                    .$(", error=").$(ex.getFlyweightMessage())
+                    .$(", errno=").$(ex.getErrno()).I$();
+            return getMin() < maxTxn;
+        }
+    }
+
+    @Override
+    public boolean incrementTxn(int id, long txn) {
+        assert txn > -1;
+        final long internalTxn = toInternalTxn(txn);
+        return incrementTxnScoreboardMem(mem, internalTxn);
+    }
+
+    @Override
+    public boolean isMax(long txn) {
+        // Unknown, return false as default
+        return false;
```

Comment:
nit: v1 scoreboard has max field internally, so we could make it public and use it here.

Reply:
We will delete V1 after the successful usage of V2, so I wanted to avoid any incompatible API calls in tests in the prod code.